Few Things

GamingLuke1337 opened the pull request:
# Changelog

Following the glorious tradition of [Keep a Changelog](https://keepachangelog.com/en/1.1.0/)
---

### Added

* Added `master_server` from [Itokoyamato's GitHub](https://github.com/Itokoyamato/TokoVOIP_TS3/tree/feat-master-server/master_server). Thanks for the Upload again!

---

### Changed

* Update check now powered by `ox_lib`
* Gave the server console print a little design refresh. It looks nicer, promise.
* Revamped the config file design
* Updated `fxmanifest` to the `cerulean` FXv2 version. More details [here](https://docs.fivem.net/docs/scripting-reference/resource-manifest/resource-manifest/#fxv2-versions).

---

Hopefully didn't break anything... but if it did, feel free to yell at me.

GamingLuke1337 committed:
Update config.lua

[Added]
- config.update

[Changed]
- Config design optimized for readability

GamingLuke1337 committed:
Update fxmanifest.lua

GamingLuke1337 committed:
Update sv_main.lua

[Added]
- ox_lib Update Check

[Removed]
- Previous Update Check method

[Changed]
- console print design

GamingLuke1337 committed:
Create package.json

GamingLuke1337 committed:
Create index.js

GamingLuke1337 committed:
Create docker-compose.yml

GamingLuke1337 committed:
Create Dockerfile

GamingLuke1337 committed:
Create README.md

GamingLuke1337 committed:
Create version.json

Plactrix reviewed (changes requested):
Hey!
Glad to see the project is still alive :)

I took a look at your commits:
- I love the new config file design, but I did see a couple of minor typos:
  - "8" instead of "("
 https://github.com/Plactrix/TokoVoIP_v2/blob/c513891d9cfe5e48e0601870affc1c1baa4d8207/fivem_script/tokovoip_script/config.lua#L47

   - I would change this to `Minimum required TS plugin version` since right now, based on the comment alone, it makes it seem like you need to have that specific version (it seems stupid, I know, but some people don't exactly read).
  https://github.com/Plactrix/TokoVoIP_v2/blob/c513891d9cfe5e48e0601870affc1c1baa4d8207/fivem_script/tokovoip_script/config.lua#L9

  - I would also probably change this to `Uncomment to enable Theater Mode` (also changed spelling since most users are American).
  https://github.com/Plactrix/TokoVoIP_v2/blob/c513891d9cfe5e48e0601870affc1c1baa4d8207/fivem_script/tokovoip_script/config.lua#L18

  - Since these three are in one block now, it's probably a good idea to clarify `TS channel password for the In-Game channel (Optional)` or something like that (again, sounds stupid, but some people aren't entirely the smartest...)
  https://github.com/Plactrix/TokoVoIP_v2/blob/c513891d9cfe5e48e0601870affc1c1baa4d8207/fivem_script/tokovoip_script/config.lua#L45-L47

  - Maybe change to something like `Use with keySwitchChannels for dual-keypress, or set to false to disable`
  https://github.com/Plactrix/TokoVoIP_v2/blob/c513891d9cfe5e48e0601870affc1c1baa4d8207/fivem_script/tokovoip_script/config.lua#L29

  - `XxX.XxX.XxX.XxX.XXXXX` --> `XxX.XxX.XxX.XxX:XXXXX`
  https://github.com/Plactrix/TokoVoIP_v2/blob/c513891d9cfe5e48e0601870affc1c1baa4d8207/fivem_script/tokovoip_script/config.lua#L38
- For `ox_lib`, since it's only being used for version checking, I don't think it should used for TokoVoIP. A lot of servers do have already ox_lib, but for those who don't have it and want to use TokoVoIP would need to add it, which is just adding unnecessary load. 
  
  Wrapping the version check function in the original code in an if statement for the config option would be fine, or even a separate file like [Badger_Discord_API](https://github.com/JaredScar/Badger_Discord_API/blob/main/versionChecker.lua) has it.

Other than that, it looks good (although I haven't tested it or anything yet)

GamingLuke1337 committed:
Update config.lua

GamingLuke1337 committed:
Update fxmanifest.lua

GamingLuke1337 committed:
Update config.lua

GamingLuke1337 committed:
Update sv_main.lua

GamingLuke1337 committed:
Create update.lua

GamingLuke1337 committed:
Update LICENSE.md

GamingLuke1337 committed:
Update fxmanifest.lua

GamingLuke1337 committed:
Update index.html

GamingLuke1337 committed:
Add files via upload

GamingLuke1337 commented:
i also felt like making the HTML a bit prettier

GamingLuke1337 committed:
Create lint.yml

GamingLuke1337 committed:
Replaced teamspeak-plugin-radiofx submodule with new GitHub repo

GamingLuke1337 committed:
add ws_server into tokovoip_script

GamingLuke1337 committed:
commit

GamingLuke1337 committed:
Update package.json

GamingLuke1337 committed:
commit

GamingLuke1337 committed:
Merge branch 'master' of https://github.com/GamingLuke1337/TokoVoIP_v2

GamingLuke1337 committed:
Update README.md

GamingLuke1337 committed:
Update README.md

GamingLuke1337 committed:
Update README.md

GamingLuke1337 committed:
Delete deps directory

GamingLuke1337 committed:
Update dependabot.yml

GamingLuke1337 committed:
Update dependabot.yml

GamingLuke1337 committed:
Create dependabot-auto-merge.yml

GamingLuke1337 committed:
Update dependabot.yml

GamingLuke1337 committed:
Update package.json

GamingLuke1337 committed:
Rename .github/dependabot-auto-merge.yml to .github/workflows/dependabot-auto-merge.yml

GamingLuke1337 committed:
commit

GamingLuke1337 committed:
Update dependabot-auto-merge.yml

GamingLuke1337 committed:
Update dependabot-auto-merge.yml

dependabot[bot] committed:
chore: bump console-stamp from 0.2.10 to 3.1.2 in /master_server

Bumps [console-stamp](https://github.com/starak/node-console-stamp) from 0.2.10 to 3.1.2.
- [Release notes](https://github.com/starak/node-console-stamp/releases)
- [Commits](https://github.com/starak/node-console-stamp/compare/v0.2.10...v3.1.2)

---
updated-dependencies:
- dependency-name: console-stamp
  dependency-version: 3.1.2
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

GamingLuke1337 committed:
Merge pull request #2 from GamingLuke1337/dependabot/npm_and_yarn/master_server/master/console-stamp-3.1.2

chore: bump console-stamp from 0.2.10 to 3.1.2 in /master_server

dependabot[bot] committed:
chore: bump cron from 1.8.3 to 4.3.1 in /master_server

Bumps [cron](https://github.com/kelektiv/node-cron) from 1.8.3 to 4.3.1.
- [Release notes](https://github.com/kelektiv/node-cron/releases)
- [Changelog](https://github.com/kelektiv/node-cron/blob/main/CHANGELOG.md)
- [Commits](https://github.com/kelektiv/node-cron/compare/v1.8.3...v4.3.1)

---
updated-dependencies:
- dependency-name: cron
  dependency-version: 4.3.1
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

GamingLuke1337 committed:
Merge pull request #3 from GamingLuke1337/dependabot/npm_and_yarn/master_server/master/cron-4.3.1

chore: bump cron from 1.8.3 to 4.3.1 in /master_server

dependabot[bot] committed:
chore: bump express from 4.21.2 to 5.1.0 in /master_server

Bumps [express](https://github.com/expressjs/express) from 4.21.2 to 5.1.0.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.21.2...v5.1.0)

---
updated-dependencies:
- dependency-name: express
  dependency-version: 5.1.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

GamingLuke1337 committed:
Merge pull request #5 from GamingLuke1337/dependabot/npm_and_yarn/master_server/master/express-5.1.0

chore: bump express from 4.21.2 to 5.1.0 in /master_server

GamingLuke1337 committed:
Update package.json

GamingLuke1337 committed:
Update and rename index.js to index.mjs

GamingLuke1337 committed:
Update package-lock.json

GamingLuke1337 committed:
Update Dockerfile

GamingLuke1337 committed:
commit

GamingLuke1337 committed:
update  ws_server dependencies

GamingLuke1337 committed:
update ws_server dependencies in tokovoip

GamingLuke1337 committed:
Update cpp-httplib

GamingLuke1337 committed:
Resolve submodule conflict: remove teamspeak-plugin-radiofx

GamingLuke1337 committed:
update submodules

GamingLuke1337 committed:
Merge remote-tracking branch 'origin/master' into pr-12

GamingLuke1337 committed:
Merge pull request #13 from GamingLuke1337/pr-12

Pr 12

Plactrix commented on README.md:
```diff
@@ -1,233 +1,76 @@
+## Terms & Conditions 
 
-Clone the repo and don't forget to initialize the submodules:
+* A visible **TokoVOIP watermark** must be shown in-game
```
Per the new LICENSE, this is no longer required

GamingLuke1337 committed:
Fix submodule: teamspeak-plugin-radiofx

GamingLuke1337 committed:
Fix submodule: teamspeak-plugin-radiofx

GamingLuke1337 committed:
Merge pull request #15 from GamingLuke1337/pr-14

Plactrix commented on .gitignore:
```diff
@@ -55,3 +55,4 @@ Temporary Items
 
 # VSCode
 .vscode
+/master_server/node_modules
```
Refer to https://github.com/Plactrix/TokoVoIP_v2/pull/27#discussion_r2189590206

Plactrix commented on .gitmodules:
```diff
@@ -7,9 +7,9 @@
 [submodule "ts3_plugin/deps/json"]
 	path = ts3_plugin/deps/json
 	url = https://github.com/nlohmann/json.git
-[submodule "ts3_plugin/deps/teamspeak-plugin-radiofx"]
```
Why are we removing RadioFX?

Plactrix commented on tokovoip/html/index.html:
```diff
@@ -0,0 +1,176 @@
+            TOKOVOIP SCRIPT
+         </div>
+         <center>
+            <font style="color: #ff0000;">TokoVOIP</font> is our in-game VOIP replacement for FiveM. We also require
```
Maybe change this and the ones below to `TokoVoIP` to match the rest of everything else

Plactrix commented on tokovoip/html/script.js:
```diff
@@ -389,6 +389,17 @@ function updateWsState(ws, state) {
+		const data = await response.json();
+		return data.ip;
+	} catch {
+		return 'IP not found';
```
Relying on only ipify.org makes it so that website NEEDS to be up in order for TokoVoIP to be able to function properly. Maybe instead of IP not found, try fetching the /getmyip endpoint since the ws_server will always be online if the game server is online

GamingLuke1337 replied:
ipify.org is one of the most used servers to get the public ip. it has a 100% uptime for the last 90 days

GamingLuke1337 replied:
i dont think that will present an issue

GamingLuke1337 replied:
> Maybe instead of IP not found, try fetching the /getmyip endpoint since the ws_server will always be online if the game server is online

like this?

```js
async function getPublicIP() {
	try {
		const response = await fetch('https://api.ipify.org?format=json');
		const data = await response.json();
		return data.ip;
	} catch (e) {
		// Ignore errors here and try fallback
		try {
			const res = await fetch(`http://${endpoint}/getmyip`);
			if (res.ok) {
				const ip = await res.text();
				return ip;
			}
		} catch (e2) {
			// Ignore errors here
		}
		return 'IP not found';
	}
}
```

Plactrix commented on tokovoip/server/sv_main.lua:
```diff
@@ -1,3 +1,14 @@
+--[[ 
+	^5| ^8 _____     _      __     __   ___ ____   __     ______      ^5|
```
You can get rid of the color coding since this is just a regular comment. The color codes were just for the console message

Plactrix commented on ts3_plugin/deps/cpp-httplib:
```diff
@@ -1 +1 @@
-Subproject commit 88c961f37efb0205aae418a35b354935be3ff3ff
+Subproject commit 145fc8b0215f5149ef4c1de6233951946294a9f3
```
The latest version of cpp-httplib does not work due to OpenSSL version deprecation in later versions.

For submodule version updates, use my commit as I built the plugin and tested to make sure they worked

Plactrix reviewed (changes requested):
- What's the difference between ws_server in your other commit versus the one that's implemented within the FiveM script?
- Out of curiosity, what's the thought process for renaming fivem_script to tokovoip?
- The changes in master_server and ws_server: Are they the same as https://github.com/Plactrix/TokoVoIP_v2/pull/27 ?

GamingLuke1337 committed:
fix stuff Plactrix told me to fix

GamingLuke1337 committed:
Merge branch 'master' into master

GamingLuke1337 committed:
Update ws and master

Updated dependencies and IPv6 Support for ws_server

GamingLuke1337 committed:
Merge branch 'master' into Plactrix

GamingLuke1337 committed:
Merge pull request #17 from GamingLuke1337/Plactrix

Update ws and master

GamingLuke1337 committed:
renaming

Plactrix reviewed (commented):
Testing pending

GamingLuke1337 committed:
add check if config.WSServerIP is an empty string

GamingLuke1337 committed:
fix for Lua54

shouldnt break anything hopefully

GamingLuke1337 committed:
Update

- Remove ws_server
- Remove ox_lib updatecheck

GamingLuke1337 committed:
add IPv6 advisory

GamingLuke1337 committed:
Update .gitignore

GamingLuke1337 commented:
have you found anything that i need to change?

GamingLuke1337 committed:
Update

GamingLuke1337 committed:
modified:   hydra_voice/client/main.lua
modified:   hydra_voice/client/utils.lua
modified:   hydra_voice/fxmanifest.lua
modified:   hydra_voice/server/main.lua
modified:   hydra_voice/server/update.lua
modified:   hydra_voice/server/utils.lua

GamingLuke1337 committed:
Update and rename .github/lint.yml to .github/workflows/lint.yml

GamingLuke1337 committed:
Create .luacheckrc

GamingLuke1337 committed:
modified:   hydra_voice/client/main.lua
modified:   hydra_voice/html/script.js
deleted:    ts3_package/package.ini
deleted:    ts3_package/plugins/tokovoip/walkie_talkie_16.png
deleted:    ts3_package/plugins/tokovoip_win64.dll

GamingLuke1337 committed:
Merge branch 'master' of https://github.com/HydraCode-GH/hydra_voice

GamingLuke1337 committed:
modified:   hydra_voice/client/main.lua
